src/lib/network_pool/indexed_pool.ml

```diff
@@ -46,9 +46,31 @@ type t =
       Transaction_hash.User_command_with_valid_signature.t
       Transaction_hash.Map.t
   ; size: int
-  ; constraint_constants: Genesis_constants.Constraint_constants.t }
+  ; constraint_constants: Genesis_constants.Constraint_constants.t
+  ; consensus_constants: Consensus.Constants.t
+  ; time_controller: Block_time.Controller.t }
 [@@deriving sexp_of]
 
+module Command_error = struct
```

Comment:
This change would have benefited from being its own PR. Please split this into a new PR and chain these expiration changes to it, so that this PR can be focused on the timing issue.